flytekit/image_spec/image_spec.py

```diff
@@ -91,37 +92,90 @@ def __post_init__(self):
         ]
         for parameter in parameters_str_list:
             attr = getattr(self, parameter)
-            parameter_is_None = attr is None
+            parameter_is_none = attr is None
             parameter_is_list_string = isinstance(attr, list) and all(isinstance(v, str) for v in attr)
-            if not (parameter_is_None or parameter_is_list_string):
+            if not (parameter_is_none or parameter_is_list_string):
                 error_msg = f"{parameter} must be a list of strings or None"
                 raise ValueError(error_msg)
 
+    @cached_property
+    def id(self) -> str:
```

Comment:
Let's say there are two workflows in two different directories with this image spec:

```python
image = ImageSpec(name="my_name", requirements="requirements-prod.txt")
```

where `requirements-prod.txt` is not the same. They will end up with the same `image_spec.id`, but different `image_spec.tag`.

Is it okay for an image spec to have the same id, but a different tag?

Reply:
Thinking about it more, having the same `id` and different `tag` is harder to come across because the local working directory is the context for `requirements` and `source_root`.

Reply:
yup, exactly